src/Arcus.EventGrid.Tests.Unit/Parsing/EventGridParsingTests.cs

```diff
@@ -22,20 +25,45 @@ public void Parse_ValidSubscriptionValidationEvent_ShouldSucceed()
             var eventTime = DateTimeOffset.Parse("2017-08-06T22:09:30.740323Z");
 
             // Act
-            var eventGridMessage = EventGridParser.Parse<SubscriptionValidation>(rawEvent);
+            var eventGridBatch = EventGridParser.ParseFromData<SubscriptionValidationEventData>(rawEvent);
 
             // Assert
-            Assert.NotNull(eventGridMessage);
-            Assert.NotNull(eventGridMessage.Events);
-            Assert.Single(eventGridMessage.Events);
-            var eventPayload = eventGridMessage.Events.Single();
+            Assert.NotNull(eventGridBatch);
+            Assert.NotNull(eventGridBatch.Events);
+            var eventPayload = Assert.Single(eventGridBatch.Events);
             Assert.Equal(eventId, eventPayload.Id);
             Assert.Equal(topic, eventPayload.Topic);
             Assert.Equal(subject, eventPayload.Subject);
             Assert.Equal(eventType, eventPayload.EventType);
             Assert.Equal(eventTime, eventPayload.EventTime);
             Assert.NotNull(eventPayload.Data);
-            Assert.Equal(validationCode, eventPayload.Data.ValidationCode);
+            Assert.Equal(validationCode, eventPayload.GetPayload()?.ValidationCode);
+        }
+
+        [Fact]
+        public void Parse_ValidNewCarRegisteredEvent_ShouldSucceed()
+        {
+            // Arrange
+            string eventId = Guid.NewGuid().ToString();
+            const string licensePlate = "1-TOM-337";
+            const string subject = licensePlate;
+            
+            var @event = new NewCarRegistered(eventId, subject, licensePlate);
+            var rawEvent = JsonConvert.SerializeObject(new[] {@event});
+
+            // Act
+            var eventGridBatch = EventGridParser.Parse<NewCarRegistered>(rawEvent);
+
+            // Assert
+            Assert.NotNull(eventGridBatch);
+            Assert.NotNull(eventGridBatch.Events);
+            var eventPayload = Assert.Single(eventGridBatch.Events);
+            Assert.Equal(eventId, eventPayload.Id);
+            Assert.Equal(subject, eventPayload.Subject);
+            Assert.Equal(@event.EventType, eventPayload.EventType);
+            Assert.Equal(@event.EventTime, eventPayload.EventTime);
+            Assert.NotNull(eventPayload.Data);
+            Assert.Equal(licensePlate, eventPayload.GetPayload()?.LicensePlate);
```

Comment:
Let's make sure we check if it's not null
```suggestion
            var eventInformation = eventPayload.GetPayload();
            Assert.NotNull(eventInformation);
            Assert.Equal(licensePlate, eventInformation.LicensePlate);
```